pkgs/development/libraries/aws-sdk-cpp/default.nix

```diff
@@ -94,9 +99,6 @@ stdenv.mkDerivation rec {
   ] ++ lib.optional (apis != ["*"])
     "-DBUILD_ONLY=${lib.concatStringsSep ";" apis}";
 
-  # fix build with gcc9, can be removed after bumping to current version
-  NIX_CFLAGS_COMPILE = [ "-Wno-error" ];
-
```

Comment:
This causes a build failure of stdenv on staging-next @vcunat:

```
In file included from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp: In constructor 'Aws::Utils::Crypto::HMACRAIIGuard::HMACRAIIGuard()':
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:416:41: error: 'HMAC_CTX* HMAC_CTX_new()' is deprecated: Since OpenSSL 3.0 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations8;;]
  416 |                     m_ctx = HMAC_CTX_new();
      |                             ~~~~~~~~~~~~^~
In file included from /build/source/aws-cpp-sdk-core/include/aws/core/utils/crypto/openssl/CryptoImpl.h:14,
                 from /build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:9,
                 from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/nix/store/gd8xbg6y246ri68pz4qkwdczydwm6liq-openssl-3.0.5-dev/include/openssl/hmac.h:33:33: note: declared here
   33 | OSSL_DEPRECATEDIN_3_0 HMAC_CTX *HMAC_CTX_new(void);
      |                                 ^~~~~~~~~~~~
In file included from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp: In destructor 'Aws::Utils::Crypto::HMACRAIIGuard::~HMACRAIIGuard()':
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:425:34: error: 'void HMAC_CTX_free(HMAC_CTX*)' is deprecated: Since OpenSSL 3.0 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations8;;]
  425 |                     HMAC_CTX_free(m_ctx);
      |                     ~~~~~~~~~~~~~^~~~~~~
In file included from /build/source/aws-cpp-sdk-core/include/aws/core/utils/crypto/openssl/CryptoImpl.h:14,
                 from /build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:9,
                 from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/nix/store/gd8xbg6y246ri68pz4qkwdczydwm6liq-openssl-3.0.5-dev/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
In file included from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp: In member function 'virtual Aws::Utils::Crypto::HashResult Aws::Utils::Crypto::Sha256HMACOpenSSLImpl::Calculate(const ByteBuffer&, const ByteBuffer&)':
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:450:29: error: 'int HMAC_Init_ex(HMAC_CTX*, const void*, int, const EVP_MD*, ENGINE*)' is deprecated: Since OpenSSL 3.0 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations8;;]
  450 |                 HMAC_Init_ex(m_ctx, secret.GetUnderlyingData(), static_cast<int>(secret.GetLength()), EVP_sha256(),
      |                 ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  451 |                              NULL);
      |                              ~~~~~
In file included from /build/source/aws-cpp-sdk-core/include/aws/core/utils/crypto/openssl/CryptoImpl.h:14,
                 from /build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:9,
                 from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/nix/store/gd8xbg6y246ri68pz4qkwdczydwm6liq-openssl-3.0.5-dev/include/openssl/hmac.h:43:27: note: declared here
   43 | OSSL_DEPRECATEDIN_3_0 int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
      |                           ^~~~~~~~~~~~
In file included from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:452:28: error: 'int HMAC_Update(HMAC_CTX*, const unsigned char*, size_t)' is deprecated: Since OpenSSL 3.0 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations8;;]
  452 |                 HMAC_Update(m_ctx, toSign.GetUnderlyingData(), toSign.GetLength());
      |                 ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /build/source/aws-cpp-sdk-core/include/aws/core/utils/crypto/openssl/CryptoImpl.h:14,
                 from /build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:9,
                 from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/nix/store/gd8xbg6y246ri68pz4qkwdczydwm6liq-openssl-3.0.5-dev/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
In file included from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:453:27: error: 'int HMAC_Final(HMAC_CTX*, unsigned char*, unsigned int*)' is deprecated: Since OpenSSL 3.0 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations8;;]
  453 |                 HMAC_Final(m_ctx, digest.GetUnderlyingData(), &length);
      |                 ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /build/source/aws-cpp-sdk-core/include/aws/core/utils/crypto/openssl/CryptoImpl.h:14,
                 from /build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:9,
                 from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/nix/store/gd8xbg6y246ri68pz4qkwdczydwm6liq-openssl-3.0.5-dev/include/openssl/hmac.h:47:27: note: declared here
   47 | OSSL_DEPRECATEDIN_3_0 int HMAC_Final(HMAC_CTX *ctx, unsigned char *md,
      |                           ^~~~~~~~~~
In file included from /build/source/build/aws-cpp-sdk-core/ub_core.cpp:35:
/build/source/aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp:458:31: error: 'int HMAC_CTX_reset(HMAC_CTX*)' is deprecated: Since OpenSSL 3.0 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations8;;]
  458 |                 HMAC_CTX_reset(m_ctx);
```

Reply:
Can you elaborate on how is `aws-sdk-cpp` related to building stdenv?

Reply:
If I would need to guess I would say darwin because it has the opposite of a minimal stdenv. Also I tested if this builds on x86_86-linux which it did that why I dropped the compile flag.